astropy/units/tests/test_units.py

```diff
@@ -570,27 +567,19 @@ def test_compose_no_duplicates():
     assert len(composed) == 1
 
 
-def test_long_int():
-    """
-    Issue #672
-    """
-    sigma = 10**21 * u.M_p / u.cm**2
-    sigma.to(u.M_sun / u.pc**2)
```

Comment:
The bug this guards against is specific to Python 2, so the test is no longer relevant.

Reply:
I'd tend to keep it, but make it more unit specific rather than go via `Quantity` (which converts the long int to float immediately).
```
long_int = 10**21  # larger than 2**63
converter = (u.M_p / u.cm**2).get_converter(u.M_sun / u.pc**2)
got = converter(long_int)
assert_allclose(got, 8.0, atol=0.1)
assert_allclose(got, converter(1e21))
```

Reply:
Python 2 had separate `int` and `long` classes, so it was important to test the code with both. But Python 3 only has `int`, so large and small integers are handled the same and it is not obvious to me what we would be testing here.

Reply:
We're testing correct conversion of long int to float in the converter. It is not completely obvious that we would notice otherwise if our code happened to not do this right. E.g., if our input was first fed to an  `np.array()` call, this would happily convert a short int to `np.int64` but a long int to `object`. So, this test helps ensure we get it right, and would warn us if, e.g., we ever messed up `_condition_arg` so that it does not do special treatment for `int` any more (and hence integers would in fact be passed on to a `np.array(value)` call. In that case, this test would give us a proper failure. (Indeed, it might be good to add `assert type(got) is float`.)

Reply:
Referring to the bug that happened with Python 2 is still needlessly confusing. I've turned `test_convert()` into a parametrized test instead.